src/test/SerializeTool_test.cpp

```diff
@@ -0,0 +1,593 @@
+        KeyFileGuard g(*this, subdir);
+        path const keyFile = subdir / ".ripple" / "secret-key.txt";
+
+        auto test = [&](std::string const kt,
+            boost::optional<std::string> const seed)
+        {
+            std::stringstream capture;
+            CoutRedirect coutRedirect{ capture };
+
+            doCreateKeyfile(keyFile, kt, seed);
+
+            auto key = RippleKey::make_RippleKey(keyFile);
+
+            auto known = boost::str(
+                boost::format("New ripple key created.\n"
+                "Stored in %s.\n"
+                "Key type is %s.\n"
+                "Account ID is %s.\n"
+                "\n\nThis file should be stored securely and not shared.\n\n")
+                % keyFile.string()
+                % to_string(key.keyType())
+                % toBase58(calcAccountID(key.publicKey())));
+
+            // Test that the function will not overwrite
+            try
+            {
+                doCreateKeyfile(keyFile, {}, {});
+                fail();
+            }
+            catch (std::exception const& e)
+            {
+                BEAST_EXPECT(e.what() ==
+                    std::string{ "Refusing to overwrite existing key file: " +
+                    keyFile.string()});
+            }
+
+            remove(keyFile);
+
+            BEAST_EXPECT(capture.str() == known);
+        };
+
+        test("", boost::none);
+        test("", std::string{ "masterpassphrase" });
+        test(to_string(KeyType::ed25519), boost::none);
+        test(to_string(KeyType::secp256k1), std::string{ "alice" });
+
+        // edge cases
+        {
+            // invalid keytype
+            std::stringstream capture;
+            CoutRedirect coutRedirect{ capture };
+
+            doCreateKeyfile(keyFile, "NSA special", boost::none);
+
+            BEAST_EXPECT(!exists(keyFile));
+
+            auto known = "Invalid key type: \"NSA special\"\n";
+
+            BEAST_EXPECT(capture.str() == known);
+        }
+        {
+            // empty seed
+            try
+            {
+                doCreateKeyfile(keyFile, "ed25519", std::string{ "" });
+                fail();
+            }
+            catch (std::exception const& e)
+            {
+                BEAST_EXPECT(e.what() ==
+                    std::string{ "Unable to parse seed: " });
+            }
+
+            BEAST_EXPECT(!exists(keyFile));
+        }
+    }
+
+    void testRepairKeyfile()
+    {
+        testcase("Repair keyfile");
+
+        using namespace boost::filesystem;
+        using namespace ripple;
+
+        std::string const subdir = "test_key_file";
+        KeyFileGuard g(*this, subdir);
+        path const keyFile = subdir / ".ripple" / "secret-key.txt";
+
+        {
+            std::stringstream capture;
+            CoutRedirect coutRedirect{ capture };
+            doCreateKeyfile(keyFile, {}, {});
+        }
+
+        // Not much to test directly here.
```

Comment:
You could modify the `master_seed` and check that repair updates the other fields accordingly.